src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -759,13 +788,29 @@ void BlockchainLMDB::remove_block()
   if ((result = mdb_cursor_del(m_cur_block_heights, 0)))
       throw1(DB_ERROR(lmdb_error("Failed to add removal of block height by hash to db transaction: ", result).c_str()));
 
-  if ((result = mdb_cursor_get(m_cur_blocks, &k, NULL, MDB_SET)))
+  MDB_val d;
+  if ((result = mdb_cursor_get(m_cur_blocks, &k, &d, MDB_SET)))
       throw1(DB_ERROR(lmdb_error("Failed to locate block for removal: ", result).c_str()));
+  uint8_t major_version = *(const uint8_t*)d.mv_data;
   if ((result = mdb_cursor_del(m_cur_blocks, 0)))
       throw1(DB_ERROR(lmdb_error("Failed to add removal of block to db transaction: ", result).c_str()));
 
   if ((result = mdb_cursor_del(m_cur_block_info, 0)))
       throw1(DB_ERROR(lmdb_error("Failed to add removal of block info to db transaction: ", result).c_str()));
+
+  if (major_version >= 4)
```

Comment:
L731 has `if (blk.major_version >= 3)`. I guess this one is wrong?

Reply:
Yes, I fixed only half on it :/